src/WingetCreateCLI/Commands/UpdateLocaleCommand.cs

```diff
@@ -0,0 +1,291 @@
+        {
+            Manifests updatedLocales = new Manifests()
+            {
+                SingletonManifest = null,
+                VersionManifest = null,
+                InstallerManifest = null,
+            };
+
+            List<string> defaultPromptPropertiesForUpdateLocale = new()
+            {
+                nameof(LocaleManifest.PackageName),
+                nameof(LocaleManifest.Publisher),
+                nameof(LocaleManifest.License),
+                nameof(LocaleManifest.ShortDescription),
+            };
+
+            bool localeArgumentUsed = false;
+            object selectedLocale;
+
+            do
+            {
+                if (!string.IsNullOrEmpty(this.Locale) && !localeArgumentUsed)
+                {
+                    selectedLocale = LocaleHelper.GetMatchingLocaleManifest(this.Locale, originalManifests);
+                    localeArgumentUsed = true;
+                }
+                else
+                {
+                    selectedLocale = PromptAllLocalesAsMenuSelection(originalManifests);
+                }
+
+                bool isDefaultLocale = false;
+
+                if (selectedLocale is LocaleManifest localeManifest)
+                {
+                    Logger.DebugLocalized(nameof(Resources.FieldSetToValue_Message), nameof(LocaleManifest.PackageLocale), localeManifest.PackageLocale);
+                    LocaleHelper.PromptAndSetLocaleProperties(localeManifest, defaultPromptPropertiesForUpdateLocale);
+                    updatedLocales.LocaleManifests.Add(localeManifest);
+                }
+                else if (selectedLocale is DefaultLocaleManifest defaultLocaleManifest)
+                {
+                    Logger.DebugLocalized(nameof(Resources.FieldSetToValue_Message), nameof(LocaleManifest.PackageLocale), defaultLocaleManifest.PackageLocale);
+                    LocaleHelper.PromptAndSetLocaleProperties(defaultLocaleManifest, defaultPromptPropertiesForUpdateLocale);
+                    updatedLocales.DefaultLocaleManifest = defaultLocaleManifest;
+                    isDefaultLocale = true;
+                }
+
+                Console.WriteLine();
+                if (Prompt.Confirm(Resources.UpdateAdditionalLocaleProperties_Message))
+                {
+                    if (isDefaultLocale)
+                    {
+                        PromptOptionalProperties(updatedLocales.DefaultLocaleManifest, LocaleHelper.GetUnPromptedLocalePropertyNames(updatedLocales.DefaultLocaleManifest, defaultPromptPropertiesForUpdateLocale));
+                    }
+                    else
+                    {
+                        LocaleManifest manifest = (LocaleManifest)selectedLocale;
+                        PromptOptionalProperties(manifest, LocaleHelper.GetUnPromptedLocalePropertyNames(manifest, defaultPromptPropertiesForUpdateLocale));
+                    }
+                }
```

Comment:
Since GetUnpromptedLocalePropertyNames takes in an anonymous parameter, do we need this if-else statement separation between defaultLocale and locale? Ideally the method should be able to handle both manifest types so that there's less repeated code.

Reply:
DefaultLocale has `moniker` and Locale doesn't? That feels weird to me, they should be identical...

Reply:
Nevermind, I did not realize but yes default locale differs from locale so that we can use the moniker for searching packages. It is okay as is.